### DISCUSSION TITLE
how to create a splited text on a container.

### DISCUSSION THREAD

**Discussion by EntropicPhys** [MyreMylar/pygame_gui#329] [Q&A] [answered]
Hi guys,

First of all, thanks for creating this excellent GUI library for python. I have been playing around with it for some time now, and, for aesthetic and practical purposes, I want to create a sort of list based on a text and text_entry_line. I sorted to make the list of text_entry_line for now, based on the UISelectionList; however, I have a problem with the text. Apparently, the UILabel can't be split once the scroll bar moves. 
I question whether there is a simple way to create a text with this property. I was reading the code from the UITextBox, but it is obscure to me how you guys do that. 

The basic idea is  as follows:

----------------------------
text0:  ###########   ^
text1:  ###########   ||
text2:  ###########   ||
text3:  ###########   ||
text4:  ###########   ^
----------------------------

######: means the Text_entry_line.  and ^
                                                                  || means the scroll bar in case that the number of entry text line's exceed the size of the 
                                                                     container.

Best, Entropic

**Comment by MyreMylar** [comment, marked as answer]
Like this?

![image](https://user-images.githubusercontent.com/13382426/188327683-546d308e-84c0-492a-8b8b-382aa856eea7.png)


Code:

```
import pygame

from pygame_gui import UIManager
from pygame_gui.elements import UITextEntryLine
from pygame_gui.elements import UILabel
from pygame_gui.elements import UIPanel
from pygame_gui.elements import UIScrollingContainer

pygame.init()

window_size = (800, 600)
window_surf = pygame.display.set_mode(window_size)

ui_manager = UIManager(window_size)


panel = UIPanel(pygame.Rect(50, 50, 250, 300), starting_layer_height=1, manager=ui_manager)
scrolling_container = UIScrollingContainer(pygame.Rect(0, 0, 250, 300), manager=ui_manager, container=panel)
scrolling_container.set_scrollable_area_dimensions((250, 500))

for y in range(0, 10):
    item_label = UILabel(pygame.Rect(0, y*50, 100, 30), "Item " + str(y),
                         manager=ui_manager, container=scrolling_container)
    item_editable_text = UITextEntryLine(pygame.Rect(120, y * 50, 100, 30),
                                         manager=ui_manager, container=scrolling_container)

running = True
clock = pygame.time.Clock()

while running:

    time_delta = clock.tick(60)/1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

        ui_manager.process_events(event)

    ui_manager.update(time_delta)

    ui_manager.draw_ui(window_surf)
    pygame.display.flip()

```

**Reply by EntropicPhys** [reply]
thanks! That was exactly what I wanted to create.
